### PR TITLE
Do not initialize `torch.distributed` process group if one is already initailized

### PR DESCRIPTION
**Pull request by Yard1**
# What does this PR do?

If a `torch.distributed` process group is already initialized, `TrainingArguments` will not initialize one by itself. This allows for the user to create their own custom process groups if needed.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by Yard1** [comment]
cc @sgugger

File: src/transformers/sagemaker/training_args_sm.py [hunk 1]
```diff
@@ -77,6 +77,11 @@ def __post_init__(self):
     @cached_property
     def _setup_devices(self) -> "torch.device":
         logger.info("PyTorch: setting up devices")
+        if torch.distributed.is_initialized() and self.local_rank == -1:
+            logger.warning(
+                "torch.distributed process group is initialized, but local_rank == -1. "
+                "In order to use Torch DDP, set local_rank or 'LOCAL_RANK' env var."
```
Comment (sgugger) [inline src/transformers/sagemaker/training_args_sm.py:83]:
```suggestion
                "In order to use Torch DDP, launch your script with `python -m torch.distributed.launch"
```
It's unlikely users will actually set those variables, the easiest way is to use the distributed launcher ;-)

File: src/transformers/training_args.py [hunk 2]
```diff
@@ -1022,10 +1022,15 @@ def eval_batch_size(self) -> int:
     @torch_required
     def _setup_devices(self) -> "torch.device":
         logger.info("PyTorch: setting up devices")
+        if torch.distributed.is_initialized() and self.local_rank == -1:
+            logger.warning(
+                "torch.distributed process group is initialized, but local_rank == -1. "
+                "In order to use Torch DDP, set local_rank or 'LOCAL_RANK' env var."
```
Comment (sgugger) [inline src/transformers/training_args.py:1028]: Same comment as above here.

**Review by sgugger — APPROVED** [review]
Thanks a lot for tackling this! LGTM with a tiny nit on the warning message.

File: src/transformers/sagemaker/training_args_sm.py (hunk 1 above)
Comment (Yard1) [reply src/transformers/sagemaker/training_args_sm.py]: @sgugger Done, PTAL

**Pull request merged**